### PR TITLE
Quiz testing

### PR DESCRIPTION
**Pull request by perryr16**
## Testing for Fear Quiz Results
* no entry error
* 1 fear
* Many fears

## SearchGameController refactor
* it was getting ugly in there
* created QuizSearchGameController to handle quiz results
* created SimilarSearchGameController to handle similar game buttons
* removed code related to ^^^ from SearchGameController

## 404 errors
* if the API response comes back with a status other than 200 (success) then it will show a 404-page error

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -25,6 +25,7 @@ gem 'omniauth-google-oauth2'
 gem 'omniauth'
 gem 'dotenv-rails'
 gem 'seed_dump'
+gem 'orderly'
```
Comment (perryr16) [inline Gemfile:28]: testing games show up in order of rating

File: app/controllers/game_search_controller.rb [hunk 2]
```diff
@@ -47,10 +38,10 @@ def create_redirect(game)
     if game.is_a?(String)
       flash[:error] = game 
       redirect_back(fallback_location: '/')
+    elsif game.is_a?(Integer)
+      render file: '/public/404'
```
Comment (perryr16) [inline app/controllers/game_search_controller.rb:42]: ah spooks

File: app/controllers/quiz_game_search_controller.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    else
+      games = create_game_objects(params)
+      create_redirect(games)
+    end
+  end
+
+  private 
+
+  def empty_quiz
+      boxes = params.keys.find_all {|param| param.include?("box")}
```
Comment (perryr16) [inline app/controllers/quiz_game_search_controller.rb:16]: boxes refer to the `check boxes id` from the fear quiz. just trying to pull out all the parameters that have the word `box`

File: app/controllers/similar_game_search_controller.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      create_redirect(game)
+  end
+
+  private 
+
+  def create_game_objects(params)
+    results = IScreamResults.new
+    game_name = params[:similar].to_s
+    game = results.create_game_objects(params[:similar])
+
```
Comment (perryr16) [inline app/controllers/similar_game_search_controller.rb:15]: A vert similar controller to others in the `GameSearch` category

File: app/poros/i_scream_results.rb [hunk 5]
```diff
@@ -15,31 +15,34 @@ def existing_game(game_name)
...
+    results[:data].map do |api_return|
+      data = api_return[:data]
+      create_game_related_objects(data)
+    end
   end
 
-  def seed_game_objects(params)
-    data = params[:data]
-    game = game_object(data)
+  def create_game_related_objects(data)
```
Comment (perryr16) [inline app/poros/i_scream_results.rb:35]: pulled code out fo similar methods to DRY it up

File: app/services/i_scream_service.rb [hunk 6]
```diff
@@ -30,6 +29,7 @@ def get_games_by_keywords(keywords)
   end
 
   def json_parse(response)
+    return 404 if response.status != 200
```
Comment (perryr16) [inline app/services/i_scream_service.rb:32]: 404!

File: spec/features/quiz/quiz_results_spec.rb [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    expect(current_path).to eq('/game_search')
+
+    expect(page).to have_css('.game-img', count: 10)
+    
+    expect('93.0').to appear_before('91.0')
+    expect('91.0').to appear_before('90.0')
+    expect('90.0').to appear_before('89.0')
+
+    selected_keywords = ['5406','24080','6919','3855']
```
Comment (perryr16) [inline spec/features/quiz/quiz_results_spec.rb:82]: test_city

File: app/controllers/game_search_controller.rb (hunk 2 above)
Comment (melatran) [reply app/controllers/game_search_controller.rb:42]: This refactor looks a lot better.

File: app/services/i_scream_service.rb (hunk 6 above)
Comment (melatran) [reply app/services/i_scream_service.rb:32]: Nice. I noticed that error popped up when I was playing around with the site

File: spec/features/quiz/quiz_results_spec.rb [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+    game1_keywords = game1.keywords.map(&:keyword)
+    expect((game1_keywords & selected_keywords).present?).to eq(true)
+    game2_keywords = game2.keywords.map(&:keyword)
+    expect((game2_keywords & selected_keywords).present?).to eq(true)
+
+  end
+  
+  
+end
```
Comment (melatran) [inline spec/features/quiz/quiz_results_spec.rb:94]: Nice robust testing.

File: app/controllers/quiz_game_search_controller.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      
+    games = results.games_by_keywords(fear_keywords)
+    session[:search_ids] = games.map(&:id) if games.is_a?(Array)
+  end
+
+  def create_redirect(game)
+    if game.is_a?(String)
+      flash[:error] = game 
+      redirect_back(fallback_location: '/')
+    elsif game.is_a?(Integer)
```
Comment (melatran) [inline app/controllers/quiz_game_search_controller.rb:34]: Not sure what this integer mean. Does this mean that it results in an error when the user enters an integer?

**Pull request merged**